DuckDuckGo/TabViewController.swift

```diff
@@ -925,11 +967,43 @@ extension TabViewController: WKNavigationDelegate {
         }
     }
     
+    /*
+     Some files might be previewed by webkit but in order to share them
+     we need to download them first.
+     This method stores the temporary download or clears it if necessary
+     */
+    private func setupOrClearTemporaryDownload(for navigationResponse: WKNavigationResponse) {
+        let downloadManager = AppDependencyProvider.shared.downloadsManager
+        
+        if let downloadMetaData = downloadManager.downloadMetaData(for: navigationResponse), !downloadMetaData.mimeType.isHTML {
+            let cookieStore = webView.configuration.websiteDataStore.httpCookieStore
+            temporaryDownload = downloadManager.setupDownload(navigationResponse,
+                                                              cookieStore: cookieStore,
+                                                              temporary: true)
+        } else {
+            temporaryDownload = nil
+        }
+    }
+    
     func webView(_ webView: WKWebView,
                  decidePolicyFor navigationResponse: WKNavigationResponse,
                  decisionHandler: @escaping (WKNavigationResponsePolicy) -> Void) {
-        decisionHandler(.allow)
-        url = webView.url
+      
+        guard !navigationResponse.canShowMIMEType else {
```

Comment:
I would see plain `if` to be more descriptive for the branching in logic we do here (and reserve guards more for required condition check). We need to make this altered webview loading logic super clear.